### PR TITLE
PHP: Improve enum cases

### PR DESCRIPTION
**Pull request by junichi11**
#### CC Light 
Before:

nb-php-enum-case-icon-light-cc-before

After:

nb-php-enum-case-icon-light-cc-after

#### Navigator Light
Before:

nb-php-enum-case-icon-light-navi-before

After:

nb-php-enum-case-icon-light-navi-after

#### CC Dark
Before:

nb-php-enum-case-icon-dark-cc-before

After:

nb-php-enum-case-icon-dark-cc-after

#### Navigator Dark
After:

nb-php-enum-case-icon-dark-navi-before

Before:

nb-php-enum-case-icon-dark-navi-after

### Improve the code completion documentation for enum cases

- Add the `isBacked()` method to the `CaseElement` and the `EnumCaseElement` interfaces
- Don't add values of non-backed enums to the code completion documentation
- Index whether it is an enum case of a backed enum as a boolean value
- Fix/Add unit tests
- Increment spec version

Before:

nb-php-improve-enum-case-cc-doc-before

After:

nb-php-improve-enum-case-cc-doc-after

### Check an initializer of an enum case 

- Fix the `IncorrectEnumHintErorr`
  - Cases of non-backed enum must not have a value (e.g. the following
    is a fatal error: `enum NonBacked {case EXAMPLE = 1;}`)
  - Cases of backed enum must have a value (e.g. the following is a
    fatal error: `enum Backed: int {case EXAMPLE;}`)
- Add/Fix unit tests

Before:

nb-php-check-enum-case-initializer-before

After:

nb-php-check-enum-case-initializer-after

### Improve the code completion for enum cases 

There is an error in the following example because `case B = self::` is not a complete statement.
Thus, the result of parsing doesn't recognize the enum declaration.
This means that some features don't work correctly.
(e.g. The enum declaration doesn't exist in the navigator. The code completion doesn't work.)

```php
enum Example: int {
    case A = 1;
    case B = self::
}
```

To improve this, add the `ASTErrorExpression` as a result of an enum case initializer part.

- Fix the parser
- Add unit tests

Before:

nb-php-improve-enum-case-navi-with-error-before

nb-php-improve-enum-case-cc-with-error-before

After:

nb-php-improve-enum-case-navi-with-error-after

nb-php-improve-enum-case-cc-with-error-after

### CODE REVIEW

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java [hunk 1]
```diff
@@ -289,15 +341,41 @@ private void checkStatements(List<Statement> statements, EnumDeclaration node) {
                     incorrectEnumProperties.add((FieldsDeclaration) statement);
                 } else if (statement instanceof UseTraitStatement) {
                     useTraits.put(node, (UseTraitStatement) statement);
+                } else if (statement instanceof CaseDeclaration) {
+                    checkCaseDeclaration((CaseDeclaration) statement, isBackedEnum);
+
```
Comment (tmysik) [inline php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java:346]: Nitpick: an extra empty line perhaps?

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java [hunk 2]
```diff
@@ -289,15 +341,41 @@ private void checkStatements(List<Statement> statements, EnumDeclaration node) {
...
+                    // enum Backed: int { case A;} // fatal error
+                    incorrectBackedEnumCases.add(caseDeclaration);
+                }
+            } else {
+                // enum NonBacked { case A = 1;} // fatal error
+                if (initializer != null) {
+                    incorrectNonBackedEnumCases.add(caseDeclaration);
+                }
+            }
+
```
Comment (tmysik) [inline php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java:365]: Nitpick: an extra empty line?

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java [hunk 3]
```diff
@@ -289,15 +341,41 @@ private void checkStatements(List<Statement> statements, EnumDeclaration node) {
...
+        }
```
Comment (tmysik) [inline php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java:366]: Nitpick:no empty line after this one.

**Review by tmysik — APPROVED** [review]
Really nice and useful improvement! 👍

**Comment by junichi11** [comment]
I'll fix them later. Thank you, Tomas!

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java (hunk 1 above)
Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java]: Fixed.

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java (hunk 2 above)
Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java]: Fixed.

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java (hunk 3 above)
Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/verification/IncorrectEnumHintError.java:364]: Fixed.

**Pull request merged**